src/main/java/com/commercetools/paymenttoorderprocessor/paymentcreationconfigurationmanager/PaymentCreationConfigurationManagerImpl.java

```diff
@@ -15,9 +15,10 @@
     private String[] transactionTypes;
 
     @Override
-    public boolean doesTransactionStateMatchConfiguration(final PaymentTransactionStateChangedMessage message, final Payment payment) {
+    public boolean doesTransactionStateMatchConfiguration(final PaymentTransactionCreatedOrUpdatedMessage message, final Payment payment) {
         final String transactionID = message.getTransactionId();
-        return TransactionState.SUCCESS.equals(message.getState())
+        final TransactionState stateFromMessage = message.getState();
```

Comment:
In case of a refund transaction would it trigger order creation endpoint too?

Reply:
> This will work for TransactionStateChanged message but not for transaction added and payment created messages since you would need to pull it out from the transaction field? This might be the issue of the message wrapper since it implies that state is always set.

It's ensured here that it pulls from the correct field.

Reply:
> Also, is it ensured that on any message a payment object is always provided?

If the program reaches this point where you commented, then it is ensured that payment is not null.

Reply:
> In case of a refund transaction would it trigger order creation endpoint too?

I did not try it, but refund should fire a different message type and state, so the message should be filter out. I could check it more if you think it's a problem, but in a different task as I don't see it as a part of this task and we're running out of time here.

Reply:
I think a refund and cancel the transaction state will be changed. So it will trigger the https://docs.commercetools.com/api/message-types#paymenttransactionstatechangedmessage I guess this is not something new according to this project when the config set for the AUTHORISE or CHARGE it should not create an order.

https://github.com/commercetools/commercetools-payment-to-order-processor/blob/3b515592c5e1e1042f308b402fe5eaf92495c4fb/src/main/java/com/commercetools/paymenttoorderprocessor/paymentcreationconfigurationmanager/PaymentCreationConfigurationManagerImpl.java#L15

Reply:
> It's ensured here that it pulls from the correct field.

could you please point out how.

Reply:
> > It's ensured here that it pulls from the correct field.
> 
> could you please point out how.

https://github.com/commercetools/commercetools-payment-to-order-processor/pull/71/files#diff-026967b3b2d039b83d87b8dda429a9cecb01979394c1886ed5aa6426c9c4c5d6R45-R51